doc/releases/migration-guide-3.5.rst

```diff
@@ -42,3 +42,42 @@ Recommended Changes
   :kconfig:option:`CONFIG_GIC_V3` directly in Kconfig has been deprecated.
   The GIC version should now be specified by adding the appropriate compatible, for
   example :dtcompatible:`arm,gic-v2`, to the GIC node in the device tree.
+
+Picolibc-related Changes
```

Comment:
Should there be something about `CBPRINTF_FP_SUPPORT`? I had to manually set it off in one of my downstream applications to match the previous firmware size and I think it's been discussed in TSC for some other application as well.

Reply:
I'm not familiar with the details of that -- can you provide some suggested wording?

Reply:
It was discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/57340#issuecomment-1551945883, maybe something on the line of picolibc providing a complete cbprintf floating point support by default at the expense of flash size, but the feature can be disabled by tweaking that flag.

Reply:
Thanks for the reminder. I've added an item which describes this issue.

Reply:
Good stuff, thanks!